internal/cmd/operator-sdk/bundle/validate/cmd.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validate
+
+import (
+	"fmt"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/operator-framework/operator-sdk/internal/cmd/operator-sdk/bundle/validate/internal"
+	"github.com/operator-framework/operator-sdk/internal/flags"
+)
+
+const (
+	longHelp = `The 'operator-sdk bundle validate' command can validate both content and format of an operator bundle
+image or an operator bundle directory on-disk containing operator metadata and manifests. This command will exit
+with an exit code of 1 if any validation errors arise, and 0 if only warnings arise or all validators pass.
+
+More information about operator bundles and metadata:
+https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md
+
+NOTE: if validating an image, the image must exist in a remote registry, not just locally.
+`
+
+	examples = `The following command flow will generate test-operator bundle manifests and metadata,
+then validate them for 'test-operator' version v0.1.0:
+
+  # Generate manifests and metadata locally.
+  $ make bundle
+
+  # Validate the directory containing manifests and metadata.
+  $ operator-sdk bundle validate ./bundle
+
+To build and validate an image built with the above manifests and metadata:
+
+  # Create a registry namespace or use an existing one.
+  $ export NAMESPACE=<your registry namespace>
+
+  # Build and push the image using the docker CLI.
+  $ docker build -f bundle.Dockerfile -t quay.io/$NAMESPACE/test-operator:v0.1.0 .
+  $ docker push quay.io/$NAMESPACE/test-operator:v0.1.0
+
+  # Ensure the image with modified metadata and Dockerfile is valid.
+  $ operator-sdk bundle validate quay.io/$NAMESPACE/test-operator:v0.1.0
+
+To list and run optional validators, which are specified by a label selector:
+
+  $ operator-sdk bundle validate --list-optional
+  NAME           LABELS                     DESCRIPTION
+  operatorhub    name=operatorhub           OperatorHub.io metadata validation
+                 suite=operatorframework
+  $ operator-sdk bundle validate --select-optional suite=operatorframework
```

Comment:
Should not the second command here has an output as well?

Reply:
Agreed I'd like to see some of the output at least.

Reply:
Done.